src/core/qgscoordinatereferencesystem.cpp

```diff
@@ -1719,7 +1721,9 @@ void QgsCoordinateReferenceSystem::setMapUnits()
          unitName.compare( QLatin1String( "hemisphere degree minute second" ), Qt::CaseInsensitive ) == 0 ||
          unitName.compare( QLatin1String( "degree (supplier to define representation)" ), Qt::CaseInsensitive ) == 0 )
       d->mMapUnits = QgsUnitTypes::DistanceDegrees;
-    else if ( unitName.compare( QLatin1String( "metre" ), Qt::CaseInsensitive ) == 0 )
+    else if ( unitName.compare( QLatin1String( "metre" ), Qt::CaseInsensitive ) == 0
```

Comment:
comparing the conversion factor (here 1.0) with let's say a 1e-10 relative error threshold would probably be more robust

Reply:
Is the conversion factor always to metres?

Reply:
For linear units, yes. And for angular units, it is to radians. If the CRS is a projected CRS, then the type of unit of the first axis is necessarily linear units. And if the CRS is a geographic CRS, the typoe of unit of the first axis is necessarily angular units.